source/_posts/angular-signals.md

```diff
@@ -0,0 +1,409 @@
+
+### Granular updates with Angular Signals
+
+With Angular Signals, there is no need to check the entire component tree and run change detection. Change detection will only be used on affected components and exactly where they have been changed.
+
+{% add_classes "Granular updates" "center bold" %}
+{% img "granular-update.png" "Granular update with Angular Signal" "lazy" %}
+
+Simply put, when the model changes, Angular will know exactly where and in which component to sync the UI without having to re-render the entire component or its tree.
+
+```ts counter.ts
+counter.set(1);	
+```
+
+```html counter.html
+<div>Count: {{ counter() }}</div> 
+// ‘Count: 1’, without re-rendering whole component
+```
+
+This seems like a great solution. So why didn't we have this in Angular before?
+
+{% img "tell-me-why.jpg" " " "lazy" %}
+
+Now we can write fully zoneless applications using **granular&nbsp;updates** of elements in the UI without re-rendering components. This opens the door to removing Zone.js from Angular altogether.
+
+What about `ChangeDetectionStrategy.OnPush`, which serves as a "smart change detection checker" that improves our application performance? Won't it be redundant when using signals?
+
+{% add_classes "OnPush change detection strategy" "center bold" %}
+{% img "change-detection-strategy-onpush.png" "OnPush change detection" "lazy" %}
+
+It seems that `OnPush` can be unnecessary. Why? Let's imagine that we don't have Zone.js and everywhere in the application we only have scattered signals and their dependencies that update themselves in a granular way. At this point, we don't need to disable any components from dirty checking in the tree anymore as we did in the `OnPush` strategy. We use signals inside and outside the components. And that's it. Everything updates automatically.
+
+## Will Angular Signals replace RxJS?
+
+As we know Angular married RxJS from the very beginning. Rxjs is the core library that is responsible for reactivity in Angular. It is a great library with which you can overcome many challenges easily and quickly. On top of that, it is very pleasant to write Angular code using:
+
+* streams,
+* [cold & hot observables](https://courses.dev-academy.com/p/advanced-rxjs),
+* subjects,
+* async pipes,
+* operators,
+* [caching](http://localhost:4000/angular-cache-http-requests/)
+* and many more.
+
+We know Angular mainly uses RxJS, so we can easily deal with asynchronicity and data passing between components. By the way, if you need to learn about the advanced techniques of RxJS then I encourage you to take a look here at this course: *[Master advanced RxJS ☄️ techniques](https://courses.dev-academy.com/p/advanced-rxjs)*.
+
+{% add_classes "Will signals replace Rxjs in some way?" "bold" %}
```

Comment:
Sometimes you write Signal with a capital letter, sometimes signal with a lower letter => please be consistent.

Reply:
Will change it to capital letter.